### ISSUE TITLE
Fitting Shift() model seems to be broken with Linear & Lev/Mar fitters

### ISSUE THREAD

**Issue by jehturner** [astropy/astropy#6103]
I might have a look at this next week. Maybe it's related to the other issues I've filed in the past week or two, where @nden commented that "most of the issues are related to constructing the derivative incorrectly or not constructing it at all" (mostly in the context of compound models):
```python
from astropy.modeling.models import *
from astropy.modeling.fitting import *
import numpy as np

x = np.arange(0,5)
z = np.array([ 0.05, 1.1, 2.08, 3.11, 4.1])

fitter = LinearLSQFitter()
model = Shift()
model_fit = fitter(model, x, z)

# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
#   File "/home/jturner/anaconda2/envs/spec/lib/python2.7/site-packages/astropy-2.0.dev18373-py2.7-linux-x86_64.egg/astropy/modeling/fitting.py", line 279, in __call__
#     lhs = model_copy.fit_deriv(x, *model_copy.parameters)
# AttributeError: 'Shift' object has no attribute 'fit_deriv'
```
This also fails in the same way with `LevMarLSQFitter()`. It works with `SLSQPLSQFitter()` and `SimplexLSQFitter()` -- but that doesn't help me, because what I _really_ want to do is fit a model set (which is only supported by the linear fitter at present, though I suppose it's not hard to implement this from scratch).

**Comment by jehturner** [comment]
I figured out how to fix this today, but now I'm running into some strange results from `LinearLSQFitter()`, not only with `Shift()` but also some other models, so it's hard to demonstrate that it works:
```python
from astropy.modeling.models import *
from astropy.modeling.fitting import *
import numpy as np

x = np.arange(0,5)
y = x+1

model = Shift()
fitter = LinearLSQFitter()
model_fit = fitter(model, x, y)
print model_fit

# Parameters:
#    offset
#    ------
#       3.0   # uh?

model = Linear1D(slope=1)
model.slope.fixed=True
fitter = LinearLSQFitter()
model_fit = fitter(model, x, y)
print model_fit

# Parameters:
#     slope intercept
#     ----- ---------
#       1.0       3.0   # huh, maybe I can't count

model = Polynomial1D(1, c1=1)
model.c1.fixed=True
fitter = LinearLSQFitter()
model_fit = fitter(model, x, y)
print model_fit

# Parameters:
#      c0  c1
#     --- ---
#     3.0 1.0   # OK, it must be true
```
Whereas the same examples with the other fitters do what one would expect:
```python
from astropy.modeling.models import *
from astropy.modeling.fitting import *
import numpy as np

x = np.arange(0,5)
y = x+1

model = Shift()
fitter = LevMarLSQFitter()
model_fit = fitter(model, x, y)
print model_fit

# Parameters:
#    offset
#    ------
#       1.0   # ah, I have a little bit of sanity left after all

model = Linear1D(slope=1)
model.slope.fixed=True
fitter = LevMarLSQFitter()
model_fit = fitter(model, x, y)
print model_fit

# Parameters:
#     slope intercept
#     ----- ---------
#       1.0       1.0

model = Polynomial1D(1, c1=1)
model.c1.fixed=True
fitter = LevMarLSQFitter()
model_fit = fitter(model, x, y)
print model_fit

# Parameters:
#      c0  c1
#     --- ---
#     1.0 1.0
```

It looks like some other models are also missing a `fit_deriv()`, eg. `Scale()`.

**Comment by bsipocz** [comment]
Fixed by #6174
